### PR TITLE
feat: add twitch helpers package

### PR DESCRIPTION
**Pull request by favna**

### CODE REVIEW

File: packages/twitch-helpers/src/lib/types.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import type { TwitchEventSubTypes, TwitchHelixBroadcasterType, TwitchHelixUserType } from './enums';
+
+export interface TwitchHelixResponse<T> {
+	data: T[];
+}
+
+export interface TwitchHelixOauth2Result {
+	access_token: string;
+
+	expires_in: number;
```
Comment (kyranet) [inline packages/twitch-helpers/src/lib/types.ts:10]: I know those newlines come from Skyra's code, but please, either document the fields with *multiline* blocks, or drop the newlines.

Style-wise, I'm fine with newlines as long as they're to separate documented fields.

File: packages/twitch-helpers/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	TwitchFetchUsersParameters,
+	TwitchHelixBearerToken,
+	TwitchHelixGameSearchResult,
+	TwitchHelixOauth2Result,
+	TwitchHelixResponse,
+	TwitchHelixStreamsResult,
+	TwitchHelixUserFollowsResult,
+	TwitchHelixUsersSearchResult,
+	TwitchOnlineEmbedData
+} from './lib/types';
```
Comment (kyranet) [inline packages/twitch-helpers/src/index.ts:18]: *Not a request*. `export type *` would be a wonderful feature to have.

TypeScript, when??

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/twitch-helpers/src/lib/twitch.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+ * const [algorithm, signature] = twitchEventSubMessageSignature.toString().split('=', 2);
+ *
+ * // Verify the signature
+ * if (!twitch.checkSignature(algorithm, signature, twitchEventSubMessage)) {
+ * 	return response.forbidden('Invalid Hub signature');
+ * }
+ * ```
+ */
+export function checkSignature(algorithm: string, signature: string, data: any) {
+	const hash = createHmac(algorithm, EventSubSecret).update(data).digest('hex');
```
[the remark this answers is not in the document]
Comment (favna) [reply packages/twitch-helpers/src/lib/twitch.ts:163]: Considering `algorithm` is based on a request header from Twitch, I'd say no. See example in tsdoc above (taken from Skyra's code) for that.

[the remark this answers is not in the document]
Comment (kyranet) [reply packages/twitch-helpers/src/lib/twitch.ts:163]: Oh, you're right.

File: packages/twitch-helpers/src/lib/types.ts (hunk 1 above)
Comment (favna) [reply packages/twitch-helpers/src/lib/types.ts:10]: They actually come from the TypeScript Interfacer Sorter vscode extension lol

File: packages/twitch-helpers/src/lib/twitch.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+			method: 'POST'
+		})
+	);
+
+	return result.match({
+		err: (error) => {
+			throw error;
+		},
+		ok: (value) => value.data[0]
+	});
```
[the remark this answers is not in the document]
Comment (favna) [reply packages/twitch-helpers/src/lib/twitch.ts:205]: Pending merge and release of sapphiredev/utilities#475

**Review by kyranet — APPROVED** [review]

**Pull request merged**